utils/build-script

```diff
@@ -1136,98 +1136,49 @@ details of the setups of other systems or automated environments.""")
             platform.system() != "Darwin"):
         args.build_foundation = True
 
-    # Propagate global --skip-build
-    if args.skip_build:
-        args.skip_build_linux = True
-        args.skip_build_freebsd = True
-        args.skip_build_cygwin = True
-        args.skip_build_osx = True
-        args.skip_build_ios = True
-        args.skip_build_tvos = True
-        args.skip_build_watchos = True
-        args.skip_build_android = True
-        args.skip_build_benchmarks = True
-        args.build_lldb = False
-        args.build_llbuild = False
-        args.build_swiftpm = False
-        args.build_xctest = False
-        args.build_foundation = False
-        args.build_libdispatch = False
+    # --validation-test implies --test.
+    if args.validation_test:
+        args.test = True
 
-    # --skip-{ios,tvos,watchos} or --skip-build-{ios,tvos,watchos} are
-    # merely shorthands for --skip-build-{**os}-{device,simulator}
-    if not args.ios or args.skip_build_ios:
+    # --long-test implies --test.
+    if args.long_test:
+        args.test = True
```

Comment:
This is not true.
See 457f2b901caad14e2085ea0fe2c9b116b5df3520 for intent of `--long-test`